server/protocols/webadmin-integration-test/webadmin-integration-test-common/src/main/java/org/apache/james/webadmin/integration/FastViewProjectionHealthCheckIntegrationTest.java

```diff
@@ -0,0 +1,220 @@
+    }
+
+    @Test
+    public void checkShouldReturnHealthyAfterSendingMessagesWithoutReading() {
+        IntStream.rangeClosed(1, 5)
+            .forEach(counter -> bobSendAMessageToAlice());
+        calmlyAwait.until(() -> listMessageIdsForAccount(aliceAccessToken).size() == 5);
+
+        webAdminApi.when()
+            .get("/healthcheck/checks/" + MESSAGE_FAST_VIEW_PROJECTION)
+        .then()
+            .statusCode(HttpStatus.OK_200)
+            .body("componentName", equalTo(MESSAGE_FAST_VIEW_PROJECTION))
+            .body("escapedComponentName", equalTo(MESSAGE_FAST_VIEW_PROJECTION))
+            .body("status", equalTo(ResultStatus.HEALTHY.getValue()));
+    }
+
+    @Test
+    public void checkShouldReturnHealthyAfterSendingAMessageWithReads() {
+        bobSendAMessageToAlice();
+        calmlyAwait.until(() -> listMessageIdsForAccount(aliceAccessToken).size() == 1);
+
+        IntStream.rangeClosed(1, 5)
+            .forEach(counter -> aliceReadLastMessage());
+
+        webAdminApi.when()
+            .get("/healthcheck/checks/" + MESSAGE_FAST_VIEW_PROJECTION)
+        .then()
+            .statusCode(HttpStatus.OK_200)
+            .body("componentName", equalTo(MESSAGE_FAST_VIEW_PROJECTION))
+            .body("escapedComponentName", equalTo(MESSAGE_FAST_VIEW_PROJECTION))
+            .body("status", equalTo(ResultStatus.HEALTHY.getValue()));
+    }
+
+    @Test
+    public void checkShouldReturnDegradedAfterFewReadsOnAMissedProjection() {
+        bobSendAMessageToAlice();
+        calmlyAwait.until(() -> listMessageIdsForAccount(aliceAccessToken).size() == 1);
+        guiceJamesServer.getProbe(JmapGuiceProbe.class)
+            .clearMessageFastViewProjection();
+
+        IntStream.rangeClosed(1, 3) // Will miss at the first time as we cleared the preview
+            .forEach(counter -> aliceReadLastMessage());
+
+        webAdminApi.when()
+            .get("/healthcheck/checks/" + MESSAGE_FAST_VIEW_PROJECTION)
+        .then()
+            .statusCode(HttpStatus.OK_200)
+            .body("componentName", equalTo(MESSAGE_FAST_VIEW_PROJECTION))
+            .body("escapedComponentName", equalTo(MESSAGE_FAST_VIEW_PROJECTION))
+            .body("status", equalTo(ResultStatus.DEGRADED.getValue()))
+            .body("cause", equalTo("retrieveMissCount percentage 25.0% (1/4) is higher than the threshold 10.0%"));
+    }
+
+    @Test
+    public void checkShouldTurnFromDegradedToHealthyAfterMoreReadsOnAMissedProjection() {
+        bobSendAMessageToAlice();
+        calmlyAwait.until(() -> listMessageIdsForAccount(aliceAccessToken).size() == 1);
+        guiceJamesServer.getProbe(JmapGuiceProbe.class)
+            .clearMessageFastViewProjection();
+
+        IntStream.rangeClosed(1, 3) // Will miss at the first time as we cleared the preview
+            .forEach(counter -> aliceReadLastMessage());
+
+        webAdminApi.when()
+            .get("/healthcheck/checks/" + MESSAGE_FAST_VIEW_PROJECTION)
+        .then()
```

Comment:
these tests break the Given-When-Then structure of a good test.
Why do you want to assert so many things?

Reply:
> I want to show to people that at the middle of the test, the healthcheck is degraded. Otherwise, no one know how is it, maybe it is already healthy, then my assertion at final become useless.

Isn't this what the first test is for ?

Reply:
> I want to show to people that at the middle of the test, the healthcheck is degraded. Otherwise, no one know how is it, maybe it is already healthy, then my assertion at final become useless.

What about writing

```
makeSystemDegraded();
```

I think the reader would understand ...